synthetic-tests/small/Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM gcc:13.2.0-bookworm AS build
+
+WORKDIR /src
+
+COPY ./barrier /src/barrier
+COPY ./dir /src/dir
+COPY ./open /src/open
+COPY ./socket /src/socket
+COPY ./stat /src/stat
+COPY ./sysconf /src/sysconf
+COPY ./utils /src/utils
+COPY ./build.sh /src/
+
+RUN ./build.sh

```

Comment:
Replace `build.sh` with another `Makefile`. Rules in the `Makefile` enter each directory and do a build (or clean).